群晖使用ALIST挂载115,无法复制文件到NAS本地盘，之前版本正常。

### Please make sure of the following things

- [x] I have read the [documentation](https://alistgo.com).
我已经阅读了[文档](https://alistgo.com)。

- [x] I'm sure there are no duplicate issues or discussions.
我确定没有重复的issue或讨论。

- [ ] I'm sure it's due to `AList` and not something else(such as [Network](https://alistgo.com/faq/howto.html#tls-handshake-timeout-read-connection-reset-by-peer-dns-lookup-failed-connect-connection-refused-client-timeout-exceeded-while-awaiting-headers-no-such-host) ,`Dependencies` or `Operational`).
我确定是`AList`的问题，而不是其他原因（例如[网络](https://alistgo.com/zh/faq/howto.html#tls-handshake-timeout-read-connection-reset-by-peer-dns-lookup-failed-connect-connection-refused-client-timeout-exceeded-while-awaiting-headers-no-such-host)，`依赖`或`操作`）。

- [ ] I'm sure this issue is not fixed in the latest version.
我确定这个问题在最新版本中没有被修复。


### AList Version / AList 版本

V3.58.0

### Driver used / 使用的存储驱动

115

### Describe the bug / 问题描述

群晖使用ALIST挂载115,无法复制文件到NAS本地盘，之前版本正常。

### Reproduction / 复现链接

115

### Config / 配置

115

### Logs / 日志

_No response_

我也遇到了，挂载后通过Alist页面能正常浏览和播放视频；
通过WebDAV挂载到Cloud Sync，能正常挂载到群晖里，但是下载到NAS会显示：下载失败。权限不足。
尝试过将WebDAV改成本地代理（默认是302重定向），但是改成本地代理后，Cloud Sync下载会无限卡住，一直显示同步中
群晖DSM版本：DSM 7.3.2-86009 Update 3
Alist套件版本： 3.58.0
302重定向模式，群晖Cloud Sync在后台的日志（已脱敏）：

> 2026-03-29T00:02:52+08:00 <HOST> syno-cloud-syncd[PID]: [ERROR] webdav-protocol.cpp(125): Server error: status='403', msg='Forbidden'
> 2026-03-29T00:02:52+08:00 <HOST> syno-cloud-syncd[PID]: [ERROR] webdav-client-handler.cpp(576): WebDAVHandler: Failed to download file from '/<REMOTE_PATH>/The.Pitt.S01E15...ass' with error='Forbidden', http_code='403'
> 2026-03-29T00:02:52+08:00 <HOST> syno-cloud-syncd[PID]: [INFO] worker.cpp(770): Erase resume info (other). [<TASK_ID>|<SUB_ID>|/<LOCAL_PATH>/The.Pitt.S01E15...ass]
> 2026-03-29T00:02:52+08:00 <HOST> syno-cloud-syncd[PID]: [ERROR] worker.cpp(2415): Worker (<WORKER_ID>): Failed to RunDownloadRemoteProtocol (-39) '/<LOCAL_PATH>/The.Pitt.S01E15...ass'

本地代理模式，报错在Alist后台：

> ERRO[2026-03-28 20:23:51] GET /dav/115_2/TV Series/.../The.Pitt.S01E15...ass webdav proxy error: 
> http request [https://cdnfhnfile.115cdn.net/<RESOURCE_ID>/The.Pitt.S01E15...ass?t=<TIMESTAMP>&u=<USER_ID>&s=<SIZE>&d=<TOKEN>&c=0&f=3&k=<SIGNATURE>&us=<LIMIT>&uc=<COUNT>&v=1] 
> failure,status: 403 
> response:{"status":403,"message":"no cookie value","request_id":"<REQUEST_ID>"}

> 我也遇到了，挂载后通过Alist页面能正常浏览和播放视频； 通过WebDAV挂载到Cloud Sync，能正常挂载到群晖里，但是下载到NAS会显示：下载失败。权限不足。 尝试过将WebDAV改成本地代理（默认是302重定向），但是改成本地代理后，Cloud Sync下载会无限卡住，一直显示同步中 群晖DSM版本：DSM 7.3.2-86009 Update 3 Alist套件版本： 3.58.0 302重定向模式，群晖Cloud Sync在后台的日志（已脱敏）：
> 
> > 2026-03-29T00:02:52+08:00  syno-cloud-syncd[PID]: [ERROR] webdav-protocol.cpp(125): Server error: status='403', msg='Forbidden'
> > 2026-03-29T00:02:52+08:00  syno-cloud-syncd[PID]: [ERROR] webdav-client-handler.cpp(576): WebDAVHandler: Failed to download file from '/<REMOTE_PATH>/The.Pitt.S01E15...ass' with error='Forbidden', http_code='403'
> > 2026-03-29T00:02:52+08:00  syno-cloud-syncd[PID]: [INFO] worker.cpp(770): Erase resume info (other). [<TASK_ID>|<SUB_ID>|/<LOCAL_PATH>/The.Pitt.S01E15...ass]
> > 2026-03-29T00:02:52+08:00  syno-cloud-syncd[PID]: [ERROR] worker.cpp(2415): Worker (<WORKER_ID>): Failed to RunDownloadRemoteProtocol (-39) '/<LOCAL_PATH>/The.Pitt.S01E15...ass'
> 
> 本地代理模式，报错在Alist后台：
> 
> > ERRO[2026-03-28 20:23:51] GET /dav/115_2/TV Series/.../The.Pitt.S01E15...ass webdav proxy error:
> > http request [https://cdnfhnfile.115cdn.net/<RESOURCE_ID>/The.Pitt.S01E15...ass?t=&u=<USER_ID>&s=&d=&c=0&f=3&k=&us=&uc=&v=1]
> > failure,status: 403
> > response:{"status":403,"message":"no cookie value","request_id":"<REQUEST_ID>"}

补充，版本回退到V3.57.0后，问题消失

--- a/drivers/115/driver.go
+++ b/drivers/115/driver.go
@@ -2,6 +2,7 @@ package _115
 
 import (
 	"context"
+	"net/http"
 	"strings"
 	"sync"
 
@@ -77,6 +78,16 @@ func (d *Pan115) Link(ctx context.Context, file model.Obj, args model.LinkArgs) 
 	link := &model.Link{
 		URL:    downloadInfo.Url.Url,
 		Header: downloadInfo.Header,
+	}
+	// The 115 CDN requires the account Cookie on the download request, but the
+	// SDK sends cookies at the transport layer so they are absent from
+	// downloadInfo.Header. Without it, proxied downloads fail with
+	// 403 "no cookie value". Restore the Cookie so proxy mode works.
+	if link.Header == nil {
+		link.Header = http.Header{}
+	}
+	if link.Header.Get("Cookie") == "" {
+		link.Header.Set("Cookie", d.Cookie)
 	}
 	return link, nil
 }

